### ISSUE TITLE
NewType/NamedTuple local inferred to be `Any` if type defined after function. 

### ISSUE THREAD

**Issue by cristicbz** [python/mypy#5515]
* **Minimal repro:**
```python
from typing import NewType


def before() -> None:
    bar = Bar(10)
    annotated_bar = Bar(10)  # type: Bar

    reveal_type(bar)  # Revealed type is 'Any'
    reveal_type(annotated_bar)  # Revealed type is 'x.Bar'


Bar = NewType('Bar', int)


def after() -> None:
    bar = Bar(10)
    reveal_type(bar)  # Revealed type is 'x.Bar'
```

* **What is the actual behavior/output?**
The type of `bar` inside `before` is inferred as `Any`.

* **What is the behavior/output you expect?**
The type of `bar` inside `before` should be inferred as `Bar`, the `NewType`.

* **What are the versions of mypy and Python you are using?**

```
(env-rw) fl•~/r/rw(dis⚡)» pip freeze | rg mypy
mypy==0.620
(env-rw) fl•~/r/rw(dis⚡)» python -V
Python 3.6.5
```

Issue is still present on master:

```
(env-rw) fl•~/r/rw(dis⚡)» pip install git+git://github.com/python/mypy.git
[...snip...]
(env-rw) fl•~/r» pip freeze | rg mypy
mypy==0.630+dev.402d734c4b8ceffdc04478eb49fc196dd2a3a785
(env-rw) fl•~/r» mypy  x.py
x.py:8: error: Revealed type is 'Any'
x.py:9: error: Revealed type is 'x.Bar'
x.py:17: error: Revealed type is 'x.Bar'
```

**Comment by ilevkivskyi** [comment]
Thanks for reporting!

We have seen similar bugs previously, but this one is a false negative and there is a simple workaround (move definition before the function), so this is probably a relatively low priority.

**Comment by JelleZijlstra** [comment]
I now get 
```
main.py:8: note: Revealed type is "__main__.Bar"
main.py:9: note: Revealed type is "__main__.Bar"
main.py:17: note: Revealed type is "__main__.Bar"
```
So this is fixed
